apps/condo/domains/property/components/panels/Builder/MapConstructor.ts

```diff
@@ -924,21 +962,21 @@ class MapEdit extends MapView {
         preview?: boolean
     ): void {
         const section = this.sections[sectionIndex]
-
         // Handle min floor adjustment
         const currentMinFloor = this.getSectionMinFloor(sectionIndex)
         const floorShift = newMinFloor - currentMinFloor
 
         if (floorShift !== 0) {
             section.floors.forEach(floor => floor.index += floorShift)
         }
-
         // Handle max floor adjustment
         const currentMaxFloor = this.getSectionMaxFloor(sectionIndex)
-        const floorDifference = newMaxFloor - currentMaxFloor
+        const currentFloorCount = this.sections[sectionIndex].floors.length
+
+        const floorDifference = newMaxFloor - currentFloorCount + (newMinFloor < 0 ? -newMinFloor : 0)
 
         if (floorDifference > 0) {
-            this.addFloorsToSection(sectionIndex, currentMaxFloor, newMaxFloor, unitsPerFloor, renameNextUnits, preview)
+            this.addFloorsToSection(sectionIndex, currentMaxFloor, (newMinFloor < 0 ? newMaxFloor - 1  : newMaxFloor), unitsPerFloor, renameNextUnits, preview)
```

Comment:
(newMinFloor < 0 ? newMaxFloor - 1  : newMaxFloor)
What is it? Put it in a constant with a clear name.